File: TP07FragmentViewPager/app/src/main/java/com/hsr2024/tp07fragmentviewpager/MyAdapter.java
package com.hsr2024.tp07fragmentviewpager;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {

    Context context;
    ArrayList<Items> items;

    public MyAdapter(Context context, ArrayList<Items> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater= LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.my_pager,parent,false);

        VH holder= new VH(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        Items item= items.get(position);

        holder.img.setImageResource(item.img);
        holder.tv.setText(item.text);


    }

    @Override
    public int getItemCount() {
        return items.size();
    }


    class VH extends RecyclerView.ViewHolder {
        ImageView img;
        TextView tv;
        public VH(@NonNull View itemView) {
            super(itemView);
            img= itemView.findViewById(R.id.iv);
            tv= itemView.findViewById(R.id.tv);
        }
    }
}

File: Ex46KotlinRecyclerViewApp/app/src/main/java/com/hsr2024/ex46kotlinrecyclerviewapp/MyAdapter.kt
package com.hsr2024.ex46kotlinrecyclerviewapp

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.bumptech.glide.Glide

class MyAdapter constructor(var context:Context,
                            var itemList:MutableList<Item>): Adapter<MyAdapter.VH>() {




    inner class VH constructor(itemView:View) : ViewHolder(itemView){
        val tvTitle:TextView = itemView.findViewById(R.id.tv_title)
        val tvMsg:TextView = itemView.findViewById(R.id.tv_msg)
        val iv:ImageView = itemView.findViewById(R.id.iv)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val inflater:LayoutInflater = LayoutInflater.from(context)
        val itemView:View = inflater.inflate(R.layout.recycler_item,parent,false)
        return VH(itemView)
    }

    //함수의 단순화...
    override fun getItemCount(): Int = itemList.size

    override fun onBindViewHolder(holder: VH, position: Int) {
        //val item:Item= itemList.get(position)
        val item:Item= itemList[position] //마치 배열처럼... 권장

        holder.tvTitle.setText(item.title)
        holder.tvMsg.text= item.message //권장

        //image load library 사용해보기 -Glide
        Glide.with(context).load(item.imgId).into(holder.iv)

    }
}